File: solutions_1483485_0/C++/vigneshswiki/tongue.cpp
#include<fstream.h>
#include<iostream.h>
#include<ctype.h>
int main()
{
    ofstream fout("C://wiki//tongue//out.out");
    ifstream fin("C://wiki//tongue//a.in");
    char tru[3][200]={"our language is impossible to understand",
                          "there are twenty six facttrual possibilities",
                          "so it is okay if you want to just give up"};
    char google[3][200]={"ejp mysljylc kd kxveddknmc re jsicpdrysi",
                          "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd",
                          "de kr kd eoya kw aej tysr re ujdr lkgc jv"};
    char alphamap[30];
    int i,j,k;
    for(i=0;i<30;i++)
                     alphamap[i]=0;
    alphamap[16]='z';
    alphamap[25]='q';
    for(i=0;i<3;i++)
    {
                    for(j=0;google[i][j];j++)
                    {
                                          if(google[i][j]==' ')
                                                          continue;
                                          alphamap[google[i][j]-97]=tru[i][j];
                                          
                    }
    }
   
    char g[30][200],english[30][200],tmp;
    int t;
    fin>>t;
    fin>>tmp;
    for(i=0;i<t;i++)
    {
                    fin.getline(g[i],200);
    }
    for(i=0;i<t;i++)
    {
                    
                    
                    for(j=0;g[i][j];j++)
                    {
                                        cout<<g[i][j];
                                        if(g[i][j]==' ')
                                                      english[i][j]=' ';
                                        else
                                        if(g[i][j]-97<27&&g[i][j]>=0)
                                        {
                                          english[i][j]=alphamap[g[i][j]-97];
                                          }
                    }
                    english[i][j]=0;
                    
                    fout<<"Case #"<<i+1<<": ";
                    if(!i)
                          fout<<alphamap[tmp-97]<<english[i]<<endl;
                    else        fout<<english[i]<<endl;
                    
                    
    }
}

File: solutions_1483485_0/java/Homero/Tongue.java
package br.com.feasoft.jam;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * User: Homer
 * Date: 4/1/12
 * Time: 9:52 PM
 */
public class Tongue {

    public static void main(String[] args) throws IOException {
        Map<Character, Character> dictionary = initDictionary();
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int numberOfCases = Integer.parseInt(reader.readLine());
        for (int i = 0; i < numberOfCases; i++) {
            String line = reader.readLine();
            System.out.printf("Case #%d: ", i + 1);
            for (int j = 0; j < line.length(); j++) {
                System.out.print(dictionary.get(line.charAt(j)));
            }
            System.out.println();
        }
    }

    private static Map<Character, Character> initDictionary() {
        Map<Character, Character> dictionary = new HashMap<Character, Character>();
        String[] googlerese = new String[4];
        String[] english= new String[4];
        
        googlerese[0] = "y qee z";
        english[0]    = "a zoo q";
        
        googlerese[1] = "ejp mysljylc kd kxveddknmc re jsicpdrysi";
        english[1]    = "our language is impossible to understand";
        
        googlerese[2] = "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd";
        english[2]    = "there are twenty six factorial possibilities";
        
        googlerese[3] = "de kr kd eoya kw aej tysr re ujdr lkgc jv";
        english[3]    = "so it is okay if you want to just give up";

        for (int i = 0; i <4; i++) {
            for (int j=0; j < googlerese[i].length(); j++) {
                dictionary.put(googlerese[i].charAt(j), english[i].charAt(j));
            }
        }
        return dictionary;
    }

}
